src/phoenix/metrics/__init__.py

```diff
@@ -1,22 +1,62 @@
-from typing import Any, Iterable, Iterator, Mapping, Protocol, runtime_checkable
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Iterable, Iterator, List, Mapping, Optional, Union
 
+import numpy as np
 import pandas as pd
 
+from phoenix.core.model_schema import Column
 
-@runtime_checkable
-class Metric(Protocol):
-    def __call__(self, df: pd.DataFrame) -> Any:
-        ...
 
+@dataclass(frozen=True)
+class Metric(ABC):
```

Comment:
For my education sake, how has the inheritance changed here - why is Protocol no longer needed / desired?